### PR TITLE
New test/test target teams distribute default shared

### PR DESCRIPTION
**Pull request by jhdavis8**
These tests ensure that a variable without explicit data-sharing attributes is shared by default(shared). It checks that both atomic updates and concurrent reads reflect that the variable is shared, and that a synchronized change to the variable by one thread is seen by all threads.

Both tests pass all compilers on Summit.

### CODE REVIEW

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90 [hunk 1]
```diff
@@ -0,0 +1,84 @@
+!===--- test_target_teams_distribute_default_shared.F90---------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses the default(shared) clause on a target teams distribute
+! directive.  The test aims to validate that when the default(shared) clause
+! is present, that all variables without explicit data sharing attributes
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90]: "that" not needed

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90 [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    share = 0
+
+    DO x = 1, N
+       a(x) = x
+    END DO
+
+    !$omp target data map(to: a(1:N)) map(tofrom: share, num_teams)
+    !$omp target teams distribute default(shared) defaultmap(tofrom:scalar)
+    DO x = 1, N
+       num_teams = omp_get_num_teams()
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90]: Data race, guard num_teams

Comment (josemonsalve2) [reply tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90]: Num teams is not used afterwards

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
+//===--- test_target_teams_distribute_default_shared.c-----------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// This test uses the default(shared) clause on a target teams distribute
+// directive.  The test aims to validate that when the default(shared) clause
+// is present, that all variables without explicit data sharing attributes
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c]: "that" can be removed

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  
+  for (int x = 0; x < N; ++x) {
+    a[x] = x;
+  }
+  
+#pragma omp target data map(to: a[0:N]) map(tofrom: share, num_teams)
+  {
+#pragma omp target teams distribute default(shared) defaultmap(tofrom:scalar)
+    for (int x = 0; x < N; ++x) {
+      num_teams = omp_get_num_teams();
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c]: data race on `num_teams`

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  for (int x = 0; x < N; ++x) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, (a[x] - 5 != x));
+    if (a[x] - 5 != x) {
+      break;
+    }
+  }
+
+  if (num_teams == 1) {
+    OMPVV_WARNING("Test operated on one team, results of default shared test are inconclusive.");
+  }
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c]: OMPVV_WARNING_IF(num_teams==1, "Test operated on one team, results of default shared test are inconclusive.");

Also add this to fortran?

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90 (hunk 2 above)
Comment (jhdavis8) [reply tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.F90]: Fixed with team num check, added warning below which uses the value.

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c (hunk 5 above)
Comment (jhdavis8) [reply tests/4.5/target_teams_distribute/test_target_teams_distribute_default_shared.c]: Added to fortran and changed to WARNING_IF

**Pull request merged**